ami/client/flowchart.py

```diff
@@ -134,10 +138,49 @@ def closeEvent(self, event):
         event.ignore()
 
 
+class BrokerFetcher(QtCore.QThread):
+
+    sig = QtCore.Signal(object)
+
+    def __init__(self, msg, broker_addr="", node=None):
+        super(__class__, self).__init__(parent=node)
+
+        self.ctx = zmq.Context()
+
+        self.node = node
+        self.broker = self.ctx.socket(zmq.SUB)
+        self.broker.connect(broker_addr)
+        self.broker.setsockopt_string(zmq.SUBSCRIBE, msg.name + ZMQ_TOPIC_DELIM)
+
+        self.running = True
+        self.sig.connect(self.node.process)
+
+    def run(self):
+        while self.running:
+            self.broker.recv_string()
+            msg = self.broker.recv_pyobj()
+
+            if isinstance(msg, fcMsgs.CloseNode):
+                self.running = False
+                self.wait()
+                self.ctx.destroy()
+            else:
+                self.sig.emit(msg)
+
+def launch_node(msg, broker_addr="", graphmgr_addr="", checkpoint_addr="",
```

Comment:
Where is this new function being used? Why is it needed?

Reply:
Oh. Its not needed. I was using that for debugging to figure out why the plot processes were crashing. Will remove it.